Realm/RLMSyncUser_Private.hpp

```diff
@@ -21,17 +21,37 @@
 #import "RLMSyncConfiguration.h"
 #import "RLMSyncUtil_Private.h"
 
-#include "sync/sync_config.hpp"
-#include "sync/impl/sync_metadata.hpp"
+#import "sync/sync_config.hpp"
+#import "sync/sync_user.hpp"
+#import "sync/impl/sync_metadata.hpp"
 
-@class RLMSyncConfiguration;
+@class RLMSyncConfiguration, RLMSyncSessionRefreshHandle;
 
 using namespace realm;
 
 typedef void(^RLMFetchedRealmCompletionBlock)(NSError * _Nullable, RLMRealm * _Nullable, BOOL * _Nonnull);
 
 NS_ASSUME_NONNULL_BEGIN
 
+class CocoaSyncUserContext : public SyncUserContext {
+public:
+    void register_refresh_handle(const std::string& path, RLMSyncSessionRefreshHandle *handle);
+    void unregister_refresh_handle(const std::string& path);
+    void invalidate_all_handles();
+
+private:
+    /**
+     A map of paths to 'refresh handles'.
+
+     A refresh handle is an object that encapsulates the concept of periodically
+     refreshing the Realm's access token before it expires. Tokens are indexed by their
+     paths (e.g. `/~/path/to/realm`).
+     */
+    std::unordered_map<std::string, RLMSyncSessionRefreshHandle *> m_refresh_handles;
+
+    mutable std::mutex m_mutex;
```

Comment:
I don't think this needs to be mutable given there are no `const` member functions.